### PR TITLE
SOLR-16714: addDepsToChanges.py handle empty gitlog

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-16714

After committing, I did a test on branch_9_2 and discovered a bug when list of solrbot changes was empty. This will detect not changes and exit script.

Also added a check to remove `(No changes)` line if it exists

### CODE REVIEW

File: dev-tools/scripts/addDepsToChanges.py [hunk 1]
```diff
@@ -102,6 +104,11 @@ def update_changes(filename, version, changes_lines):
                         buffer.append(change_line)
                         buffer.append("\n")
                     continue
+            else:
```
Comment (HoustonPutman) [inline dev-tools/scripts/addDepsToChanges.py:107]: Don't think you need an `else` here, but it doesn't really matter.

**Review by HoustonPutman — APPROVED** [review]
Looks good!

File: dev-tools/scripts/addDepsToChanges.py [hunk 2]
```diff
@@ -124,8 +125,11 @@ def main():
         gitlog_lines = run(
             'git log --author=' + newconf.user + ' --oneline --no-merges --pretty=format:"%s" ' + prev_tag + '..').split(
             "\n")
-        changes_lines = list(map(lambda l: gitlog_to_changes(l, newconf.user), gitlog_lines))
-        update_changes('solr/CHANGES.txt', newconf.version, changes_lines)
+        changes_lines = list(map(lambda l: gitlog_to_changes(l, newconf.user), list(filter(None, gitlog_lines))))
```
[the remark this answers is not in the document]
Comment (janhoy) [reply dev-tools/scripts/addDepsToChanges.py:134]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (janhoy) [reply dev-tools/scripts/addDepsToChanges.py:134]: @sonatype-lift ignore

File: dev-tools/scripts/addDepsToChanges.py (hunk 1 above)
Comment (janhoy) [reply dev-tools/scripts/addDepsToChanges.py]: Good catch. Removed it, and also tightened the requiremnet of `---` line after version header.

**Pull request merged**